### PR TITLE
[MIG] sale_cancel_reason: Migration to 13.0

### PR DESCRIPTION
**Pull request by espo-tony**

### CODE REVIEW

File: sale_cancel_reason/view/sale_order_cancel_reason_view.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record model="ir.ui.view" id="view_sale_order_cancel_reason_form">
```
Comment (sudhir-erpharbor) [inline sale_cancel_reason/view/sale_order_cancel_reason_view.xml:4]: sale_order_cancel_reason_view.xml and sale_order_view.xml both files have similar code. Please remove any one of the files.

File: sale_cancel_reason/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
```
Comment (sudhir-erpharbor) [inline sale_cancel_reason/__init__.py:1]: UTF-8 coding is not necessary

File: sale_cancel_reason/view/sale_order_cancel_reason_view.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            <tree string="Sale Order Cancel Reasons">
+                <field name="name"/>
+            </tree>
+        </field>
+    </record>
+
+    <record model="ir.actions.act_window" id="action_sale_order_cancel_reason">
+        <field name="name">Sale Order Cancel Reason</field>
+        <field name="res_model">sale.order.cancel.reason</field>
+        <field name="view_type">form</field>
```
Comment (sudhir-erpharbor) [inline sale_cancel_reason/view/sale_order_cancel_reason_view.xml:27]: view_type field has been removed from the window action. Please remove it from all the actions in XML / py.

**Comment by kittiu** [comment]
> _No description provided._

Hi, I think the name of this PR should be
`[13.0][MIG] sale_cancel_reason`

**Comment by kittiu** [comment]
1. You can add yourself as contributor in readme folder.
2. Please merge your commits to 1 commit.
3. Please go through runbot, travis, and fix them.

Thank you for your contribution!

**Comment by marylla** [comment]
Hi @espo-tony 
Can you check whats the problem with travis ci is? 
I would like to make a functional review here. :)

**Comment by newtratip** [comment]
Hello @espo-tony 

Please check travis-ci that what is the error and squash your commit.

Guide line for squah the commit
https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

Thanks!

**Comment by rousseldenis** [comment]
@espo-tony Do you mind if we close this in favor of https://github.com/OCA/sale-workflow/pull/1306 ?

File: sale_cancel_reason/wizard/cancel_reason.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        string='Reason',
+        required=True)
+
+    def confirm_cancel(self):
+        self.ensure_one()
+        act_close = {'type': 'ir.actions.act_window_close'}
+        sale_ids = self._context.get('active_ids')
+        if sale_ids is None:
+            return act_close
+        assert len(sale_ids) == 1, "Only 1 sale ID expected"
```
Comment (rousseldenis) [inline sale_cancel_reason/wizard/cancel_reason.py:23]: Please use UserError. We are in an Odoo context, so using its exceptions is better.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
@espo-tony What's the status of this ? Could you attend comments?

**Pull request closed without merging**